packages/next/build/index.ts

```diff
@@ -2216,6 +2218,15 @@ export default async function build(
       ;(images as any).sizes = [...deviceSizes, ...imageSizes]
       ;(images as any).remotePatterns =
```

Comment:
@ijjk Do we care about being backwards compatible here? Its experimental so I'm wonder if we want to keep this as-is or just break it now to always return regex?

Reply:
I think we can tell the difference between old and new format by looking for the leading `^`

Reply:
I think checking for `^` should be good, although it should also be ok to make breaking change here if needed